contracts/events/TopicEvent.sol

```diff
@@ -299,54 +292,43 @@ contract TopicEvent is ITopicEvent, BaseContract, Ownable {
     }
 
     /* 
-    * @notice Calculates the QTUM tokens won based on the sender's QTUM token contributions.
-    * @return The amount of QTUM tokens won.
+    * @notice Calculates the BOT and QTUM tokens won based on the sender's contributions.
+    * @return The amount of BOT and QTUM tokens won.
     */
-    function calculateQtumContributorWinnings() 
+    function calculateWinnings()
         public 
         view
         inCollectionStatus()
-        returns (uint256)  
-    {
-        uint256 senderContribution = balances[resultIndex].bets[msg.sender];
-        uint256 winnersTotal = balances[resultIndex].totalBets;
-        uint256 losersTotalMinusCut = 0;
-        for (uint8 i = 0; i < numOfResults; i++) {
-            if (i != resultIndex) {
-                losersTotalMinusCut = losersTotalMinusCut.add(balances[i].totalBets);
-            }
-        }
-        losersTotalMinusCut = losersTotalMinusCut.mul(90).div(100);
-        uint256 senderContributionMinusCut = senderContribution.mul(90).div(100);
-        return senderContribution.mul(losersTotalMinusCut).div(winnersTotal).add(senderContributionMinusCut);
-    }
-
-    /*
-    * @notice Calculates the QTUM and BOT tokens won based on the sender's BOT contributions.
-    * @return The amount of QTUM and BOT tokens won.
-    */
-    function calculateBotContributorWinnings()
-        public
-        view
-        inCollectionStatus()
-        returns (uint256, uint256) 
+        returns (uint256, uint256)  
     {
-        // Calculate BOT won
-        uint256 senderContribution = balances[resultIndex].votes[msg.sender];
+        // Calculate BOT winnings
+        uint256 contribution = balances[resultIndex].votes[msg.sender];
```

Comment:
Add zero contribution check to bot BOT and VOTE to exit early?

Reply:
done